### PR TITLE
Use snapcraft-preload to redirect requests to the proper target

### PR DESCRIPTION
**Pull request by 3v1n0**
This will allow to get this working also in non-ubuntu
environments, and getting rid of the unneeded startup script.

It fixes issue #95.

### CODE REVIEW

**Comment by 3v1n0** [comment]
@kyrofa have a look to this when you can, please
@oparoz it would be cool if you could generate the keys to upload this to ubuntu store

File: linux/snap/snapcraft.yaml [hunk 1]
```diff
@@ -51,7 +48,8 @@ apps:
       - nextcloud
     command: |
       env LD_LIBRARY_PATH=$SNAP/usr/lib/nextcloud:$LD_LIBRARY_PATH
-      desktop-launch nextcloud
+          FONTCONFIG_FILE=$SNAP/etc/fonts/fonts.conf
+          snapcraft-preload nextcloud
```
Comment (kyrofa) [inline linux/snap/snapcraft.yaml]: Have you tried using the `environment` keyword, here?

Comment (3v1n0) [reply linux/snap/snapcraft.yaml]: I did, and it wasn't working in the snapd versions available in trusty and xenial, it should be fixed upstream though...

Comment (kyrofa) [reply linux/snap/snapcraft.yaml]: Yeah there were some bugs there for a while. Worth checking if the fixes have been released?

**Comment by 3v1n0** [comment]
@kyrofa ok, the `environment` stanza now works in both zeisty, xenial and trusty, so we can safely use it.

Updated as requested.

**Comment by enoch85** [comment]
@kyrofa Is this OK with you? I don't have a clue about snaps, but if it looks ok to you, please review so that we can merge it.

**Comment by enoch85** [comment]
Also, *please* @oparoz can you help @3v1n0 with publishing on the store?

**Review by kyrofa — APPROVED** [review]
That looks quite nice now, thank you!

**Pull request merged**